nexus/api_test.go

```diff
@@ -74,3 +74,144 @@ func TestFailure_JSONMarshalling(t *testing.T) {
 		})
 	}
 }
+
+func TestEncodeLink(t *testing.T) {
```

Comment:
Please add a case for when the URL is invalid and prove that you generate a valid header if the URL requires quoting and such.